src/internal/operators/concatAll.ts

```diff
@@ -61,6 +61,6 @@ export function concatAll<R>(): OperatorFunction<any, R>;
  * @return {Observable} An Observable emitting values from all the inner
  * Observables concatenated.
  */
-export function concatAll<T>(): OperatorFunction<ObservableInput<T>, T> {
-  return mergeAll<T>(1);
+export function concatAll<O extends ObservableInput<any>>(): OperatorFunction<O, ObservedValueOf<O>> {
```

Comment:
Make sure to circle back with the Google folks to see how safe it is to change the generic here. Maybe keep this in mind for the next core meeting

Reply:
That's sort of the point of this major, right? To get all of the types correct. That's the breaking change.